src/content/app/tools/blast/state/blast-form/blastFormSlice.ts

```diff
@@ -253,6 +333,25 @@ const blastFormSlice = createSlice({
         }
       });
     },
+    // this action is for populating BLAST form via the "BLAST this sequence" button in another app
+    setSequenceForGenome(
+      state,
+      action: PayloadAction<{
+        sequence: ParsedInputSequence;
+        species: Species;
+        sequenceType: SequenceType;
+        config: BlastSettingsConfig;
+      }>
+    ) {
+      const { sequence, species, sequenceType, config } = action.payload;
+      resetSettings(state, config);
+      state.sequences = [sequence];
+      state.selectedSpecies = [species];
+      state.shouldAppendEmptyInput = false;
+      state.settings.sequenceType = sequenceType;
+      state.settings.sequenceSelectionMode = 'manual'; // FIXME: does this suggest the need for a better flag name?
```

Comment:
How about

Reply:
I was more concerned with the word "manual" here, which we take as a signal to prevent automatic updates to the sequence type or the database. In this reducer, we are setting sequence selection mode to "manual" although the user has not manually selected the sequence type. I was wondering to myself here whether this suggests that instead of 

```
{
  sequenceSelectionMode: 'automatic' | 'manual';
  databaseSelectionMode: 'automatic' | 'manual';
}
```

 we should instead use something like:

```
{
  autoUpdateSettings: {
    sequenceType: boolean;
    database: boolean;
  }  
}
```

Reply:
Decided to keep this as it is. Removing the FIXME.